### PR TITLE
Improve vars with refernce in Settings cpps

### PR DESCRIPTION
**Pull request by xuzifu666**
### Changelog category (leave one):

- Not for changelog

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Improve vars with refernce in Settings cpps

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by xuzifu666** [comment]
@alexey-milovidov Hi，could you give a review please

File: src/Core/BaseSettings.h [hunk 1]
```diff
@@ -469,7 +469,7 @@ void BaseSettings<TTraits>::write(WriteBuffer & out, SettingsWriteFormat format)
 {
     const auto & accessor = Traits::Accessor::instance();
 
-    for (auto field : *this)
+    for (auto & field : *this)
```
Comment (nikitamikhaylov) [inline src/Core/BaseSettings.h:472]:
```suggestion
    for (const auto & field : *this)
```

File: src/Core/ServerSettings.cpp [hunk 2]
```diff
@@ -25,7 +25,7 @@ void ServerSettings::loadSettingsFromConfig(const Poco::Util::AbstractConfigurat
         "max_remote_write_network_bandwidth_for_server",
     };
 
-    for (auto setting : all())
+    for (auto & setting : all())
```
Comment (nikitamikhaylov) [inline src/Core/ServerSettings.cpp:28]:
```suggestion
    for (const auto & setting : all())
```

File: src/Core/Settings.cpp [hunk 3]
```diff
@@ -87,7 +87,7 @@ void Settings::checkNoSettingNamesAtTopLevel(const Poco::Util::AbstractConfigura
         return;
 
     Settings settings;
-    for (auto setting : settings.all())
+    for (auto & setting : settings.all())
```
Comment (nikitamikhaylov) [inline src/Core/Settings.cpp:90]:
```suggestion
    for (const auto & setting : settings.all())
```

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit 236e5d4f11a374f7fea461b910b29acc4d799338 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54220/236e5d4f11a374f7fea461b910b29acc4d799338/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Core/Settings.cpp (hunk 3 above)
Comment (xuzifu666) [reply src/Core/Settings.cpp]: @nikitamikhaylov done，thanks~

**Comment by alexey-milovidov** [comment]
This does not give anything to the user - no new features, no changes in behavior or performance.
Changing to "Not for changelog".

**Comment by xuzifu666** [comment]
> This does not give anything to the user - no new features, no changes in behavior or performance. Changing to "Not for changelog".

@alexey-milovidov  this pr aim to avoid obj constructor call and can improve performance，so added improvement before

**Comment by alexey-milovidov** [comment]
There is no observable difference in performance in any user-visible scenario.

**Pull request merged**